app/webpacker/packs/cookie-banner.js

```diff
@@ -0,0 +1,44 @@
+const cookieBannerEl = document.querySelector(".js-cookie-banner");
+
+if (cookieBannerEl) {
+  const cookieFormEl = document.querySelector(".js-cookie-form");
+
+  cookieFormEl.addEventListener("click", (e) => {
+    if (e.target.tagName !== "BUTTON") {
+      return;
+    }
+
+    fetch(cookieFormEl.action, {
+      method: "PUT",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        cookies_form: { analytics_consent: e.target.value },
+      }),
+    })
+      .then((res) => res.json())
+      .then(({ message }) => {
+        const messageEl = cookieBannerEl.querySelector(".js-cookie-message");
+        messageEl.textContent = message;
+
+        Array.from(cookieBannerEl.children).forEach((childEl) => {
+          if (childEl.hasAttribute("hidden")) {
+            childEl.removeAttribute("hidden");
+          } else {
+            childEl.setAttribute("hidden", "");
+          }
+        });
+      });
+
+    e.preventDefault();
+  });
```

Comment:
Means if you get halfway through a form and then click the cookie banner, you don't lose your progress in the form.

It really annoys me when cookie banners refresh the page!

Reply:
I mean, do we expect users to see a cookie banner on a page with a form? And of those users who do, do we expect any of them to click on 'accept cookies' as they type their email address or something?

Also, more importantly, I think it leads to the need to hide and unhide a bunch of elements with javascript, thus removing the readability benefits of rendering things server-side. 
```
        Array.from(cookieBannerEl.children).forEach((childEl) => {
          if (childEl.hasAttribute("hidden")) {
            childEl.removeAttribute("hidden");
          } else {
            childEl.setAttribute("hidden", "");
          }
        });
```
Like, I can read what it does, and I guess it says 'hide all elements that are shown, show all elements that are hidden'?But then when I read the page, it's not at all obvious what will happen when - and if someone uses a screen reader with js enabled, I'm not sure how intuitive it will be for them :/

Reply:
I wonder how the other teams in DfE have done this. Did you do a quick check?

Reply:
This pattern is taken from the GDS style guide. The implementation they use as an example on GOV.UK Notify is client-side only. So is the one on www.gov.uk.

Reply:
Also, there is `childEl.toggleAttribute('hidden')` but it doesn't work in IE and it was quicker to just not use it than to work out if babel is adding that polyfill 🥲